### PR TITLE
packetbeat 5.0.1 (new formula)

### PR DESCRIPTION
**Pull request by ruflin**
- [x] Have you followed the [guidelines for contributing](https://github.com/Homebrew/homebrew-core/blob/master/CONTRIBUTING.md)?
- [x] Have you checked that there aren't other open [pull requests](https://github.com/Homebrew/homebrew-core/pulls) for the same formula update/change?
- [x] Have you built your formula locally with `brew install --build-from-source <formula>`, where `<formula>` is the name of the formula you're submitting?
- [x] Does your build pass `brew audit --strict <formula>` (after doing `brew install <formula>`)?

-----

There is one issue with Packetbeat: Because it access the network interface it needs to be run as root which make testing it quite tricky. Any recommendations here from the homebrew team?

### CODE REVIEW

**Comment by woodruffw** [comment]
We probably can't give a `test` block elevated permissions, so a simpler test (`--version` or `--help`) should be fine here.

**Comment by ruflin** [comment]
@woodruffw Thanks for the input. I added to simple commands to check for packetbeat. Now it looks like the builds also go green.

File: Formula/packetbeat.rb [hunk 1]
```diff
@@ -0,0 +1,55 @@
...
+      libexec.install "packetbeat"
+
+      inreplace "packetbeat.yml", "packetbeat.interfaces.device: any", "packetbeat.interfaces.device: en0"
+
+      (etc/"packetbeat").install("packetbeat.yml", "packetbeat.template.json", "packetbeat.template-es2x.json")
+    end
+
+    (bin/"packetbeat").write <<-EOS.undent
+      #!/bin/sh
+      exec #{libexec}/packetbeat -path.config #{etc}/packetbeat -path.home #{prefix} -path.logs #{var}/log/packetbeat -path.data #{var}/packetbeat $@
```
Comment (MikeMcQuaid) [inline Formula/packetbeat.rb]: Can you split this onto multiple lines with `\` at 80 characters.

File: Formula/packetbeat.rb [hunk 2]
```diff
@@ -0,0 +1,55 @@
...
+        <string>#{opt_bin}/packetbeat</string>
+        <key>RunAtLoad</key>
+        <true/>
+      </dict>
+    </plist>
+  EOS
+  end
+
+  test do
+    system "#{libexec}/packetbeat", "--version"
```
Comment (MikeMcQuaid) [inline Formula/packetbeat.rb]: Can remove this test.

File: Formula/packetbeat.rb [hunk 3]
```diff
@@ -0,0 +1,55 @@
...
+    system "#{libexec}/packetbeat", "--devices"
```
Comment (MikeMcQuaid) [inline Formula/packetbeat.rb]: Call `#{bin}/packetbeat` to test your script.

File: Formula/packetbeat.rb [hunk 4]
```diff
@@ -0,0 +1,55 @@
...
+    gopath = buildpath/"gopath"
+    (gopath/"src/github.com/elastic/beats").install Dir["{*,.git,.gitignore}"]
+
+    ENV["GOPATH"] = gopath
+
+    cd gopath/"src/github.com/elastic/beats/packetbeat" do
+      system "make"
+      libexec.install "packetbeat"
+
+      inreplace "packetbeat.yml", "packetbeat.interfaces.device: any", "packetbeat.interfaces.device: en0"
```
Comment (ruflin) [inline Formula/packetbeat.rb]: Under OS X `any` is not supported. It has to be `en0` or `lo0`. The binary package of packetbeat shipped for OS X has already `en0` in the config as we do it during packaging. But as we get it here from the source, `any` has to bereplaced by `en0`.

File: Formula/packetbeat.rb (hunk 2 above)
Comment (ruflin) [inline Formula/packetbeat.rb]: done

File: Formula/packetbeat.rb (hunk 1 above)
Comment (ruflin) [inline Formula/packetbeat.rb]: changed

File: Formula/packetbeat.rb (hunk 3 above)
Comment (ruflin) [inline Formula/packetbeat.rb]: Done

**Comment by ruflin** [comment]
Updated version pushed.

File: Formula/packetbeat.rb (hunk 4 above)
Comment (MikeMcQuaid) [inline Formula/packetbeat.rb]: Ok, thanks.

**Comment by MikeMcQuaid** [comment]
Thanks again for your contribution to Homebrew! Without people like you submitting PRs we couldn't run this project. You rock!

For future reference the preferred commit message format is `packetbeat 5.0.1 (new formula)`. Please complete the issue template in future PRs where this format is detailed in the linked [Contributing guidelines](https://github.com/Homebrew/homebrew-core/blob/master/CONTRIBUTING.md). Thanks!

**Comment by ruflin** [comment]
@MikeMcQuaid Thanks for merging. Sorry about the commit message, will do better next time.

**Pull request closed without merging**